packages/devtools_app/test/flutter/memory_screen_test.dart

```diff
@@ -0,0 +1,140 @@
+import 'package:devtools_app/src/globals.dart';
+import 'package:devtools_app/src/service_manager.dart';
+import 'package:devtools_app/src/memory/flutter/memory_chart.dart';
+import 'package:devtools_app/src/memory/flutter/memory_screen.dart';
+import 'package:devtools_app/src/memory/memory_controller.dart';
+import 'package:devtools_app/src/ui/fake_flutter/_real_flutter.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/mockito.dart';
+
+import '../support/mocks.dart';
+import 'wrappers.dart';
+
+void main() {
+  MemoryScreen screen;
+  MemoryController controller;
+  FakeServiceManager fakeServiceManager;
+
+  Future<void> pumpMemoryScreen(
+    WidgetTester tester, {
+    MemoryController memoryController,
+  }) async {
+    // Set a wide enough screen width that we do not run into overflow.
+    await tester.pumpWidget(wrapWithControllers(
+      const MemoryBody(),
+      memoryController: controller = memoryController ?? MemoryController(),
+    ));
+    expect(find.byType(MemoryBody), findsOneWidget);
+  }
+
+  const windowSize = Size(1599.0, 1000.0);
+
+  group('MemoryScreen', () {
+    setUp(() async {
+      await ensureInspectorDependencies();
+      fakeServiceManager = FakeServiceManager(useFakeService: true);
+      setGlobal(ServiceConnectionManager, fakeServiceManager);
+      when(serviceManager.connectedApp.isDartWebApp)
+          .thenAnswer((_) => Future.value(false));
+      screen = const MemoryScreen();
+    });
+
+    testWidgets('builds its tab', (WidgetTester tester) async {
+      await tester.pumpWidget(wrap(Builder(builder: screen.buildTab)));
+      expect(find.text('Memory'), findsOneWidget);
+    });
+
+    testWidgetsWithWindowSize('builds proper content for state', windowSize,
+        (WidgetTester tester) async {
+      await pumpMemoryScreen(tester);
+
+      // Should be collecting live feed.
+      expect(controller.offline, isFalse);
+
+      var splitFinder = find.byType(Split);
+
+      // Verify Memory, Memory Source, and Memory Sources content.
+      expect(splitFinder, findsOneWidget);
+      expect(find.byKey(MemoryScreen.pauseButtonKey), findsOneWidget);
+      expect(find.byKey(MemoryScreen.resumeButtonKey), findsOneWidget);
+
+      expect(find.byKey(MemoryScreen.memorySourceStatusKey), findsOneWidget);
+      final Text memorySourceText = tester.firstWidget(find.byKey(
+        MemoryScreen.memorySourceStatusKey,
+      )) as Text;
+      expect(memorySourceText.data, MemoryBodyState.liveFeed);
+
+      expect(find.byKey(MemoryScreen.snapshotButtonKey), findsOneWidget);
+      expect(find.byKey(MemoryScreen.resetButtonKey), findsOneWidget);
+      expect(find.byKey(MemoryScreen.gcButtonKey), findsOneWidget);
+
+      expect(find.byType(MemoryChart), findsOneWidget);
+
+      expect(controller.memoryTimeline.data.isEmpty, isTrue);
+      expect(controller.memoryTimeline.offflineData.isEmpty, isTrue);
```

Comment:
nit: instead write `expect(controller.memoryTimeline.data, isEmpty);` and `expect(controller.memoryTimeline.offflineData, isEmpty);`

Reply:
Done.